### PR TITLE
New tests, Windows support

### PR DESCRIPTION
**Pull request by robooo**
- Tests are now executable on Windows
- CI on Appveyor
- Merged results of coverage from Travis and Appveyor with Codecov
- Few improvements of existing tests
- Update of README

### CODE REVIEW

File: spec/config_spec.lua [hunk 1]
```diff
@@ -53,38 +62,61 @@ describe("LuaRocks config tests #blackbox #b_config", function()
    end)
 
    describe("LuaRocks config - more complex tests", function()
+      local scdir = testing_paths.testing_lrprefix .. "/etc/luarocks"
+      local versioned_scname = scdir .. "/config-" .. env_variables.LUA_VERSION .. ".lua"
+      local scname = scdir .. "/config.lua"
+
+      local configfile
```
Comment (mpeterv) [inline spec/config_spec.lua]: This is unused

File: spec/config_spec.lua [hunk 2]
```diff
@@ -53,38 +62,61 @@ describe("LuaRocks config tests #blackbox #b_config", function()
...
-         sysconfig:close()
-
-         local output = run.luarocks("config --system-config;")
-         assert.are.same(output, scdir .. "/config.lua")
-         test_env.remove_dir(testing_paths.testing_lrprefix)
+         if test_env.TEST_TARGET_OS == "windows" then
+            local output = run.luarocks("config --system-config")
+            assert.are.same(output, versioned_scname)
+         else
+            sysconfig = io.open(scname, "w+")
```
Comment (mpeterv) [inline spec/config_spec.lua]: Missing `local` here and in the other `if` branch

File: test/test_environment.lua [hunk 3]
```diff
@@ -572,8 +661,8 @@ function test_env.main()
 
    local install_env_vars = {
       LUAROCKS_CONFIG = test_env.testing_paths.testing_dir .. "/testing_config.lua",
-      LUA_PATH = "",
-      LUA_CPATH = ""
+      LUA_PATH,
```
Comment (mpeterv) [inline test/test_environment.lua]: LUA_PATH and LUA_CPATH are now globals here, is this change intended?

Comment (robooo) [inline test/test_environment.lua]: cmd on windows took as path empty space, so I rewritten it like this, maybe hisham knows another solution

Comment (mpeterv) [inline test/test_environment.lua]: The two lines after the change do not do anything, `{LUAROCKS_CONFIG = ..., LUA_PATH, LUA_CPATH}` == `{LUAROCKS_CONFIG = ...}` because `LUA_PATH` and `LUA_CPATH` are undefined globals

File: spec/config_spec.lua (hunk 1 above)
Comment (hishamhm) [inline spec/config_spec.lua]: Is it? Tests below (lines 77-79, 87-98) use configfile, which is either scname or versioned_scname, which on their turn use scdir.
